### ISSUE TITLE
MacOS Theano Error ---- ['MRG_RandomStreams' from 'theano.sandbox.rng_mrg']

### ISSUE THREAD

**Issue by CloudChaoszero** [pymc-devs/Theano-PyMC#226]
## Description of your problem or feature request

Hey team, hope you are doing well & are safe! :) 
### TL;DR
 I receive an error from Theano, typically after creating new conda environment, and running examples

The latest error:
```python

ImportError: cannot import name 'MRG_RandomStreams' from 'theano.sandbox.rng_mrg' (/anaconda3/envs/temp/lib/python3.8/site-packages/theano/sandbox/rng_mrg.py)
```
### 1.New Conda environment

```
conda create -n temp python=3.8 -y
conda activate temp
conda install -c conda-forge pymc3
```

I first installed Pymc3 through conda forge
`conda install -c conda-forge pymc3`

After, I was faced with an error quite similar to #127, when I ran this [api quickstart] piece of code in a Notebook:

#### Code
```python
import arviz as az
import matplotlib.pyplot as plt
import numpy as np
import pymc3 as pm
import theano.tensor as tt
import warnings
warnings.simplefilter(action="ignore", category=FutureWarning)


%config InlineBackend.figure_format = 'retina'
az.style.use('arviz-darkgrid')
print('Running on PyMC3 v{}'.format(pm.__version__))
print('Running on ArviZ v{}'.format(az.__version__))

with pm.Model() as model:
    mu = pm.Normal("mu", mu=0, sigma=1)
    obs = pm.Normal("obs", mu=mu, sigma=1, observed=np.random.randn(100))
```

#### Error

```python
    thunk, module = self.cthunk_factory(
  File "/Users/dforeman/src/pymc-devs/Theano-PyMC/theano/gof/cc.py", line 1720, in cthunk_factory
    module = get_module_cache().module_from_key(
  File "/Users/dforeman/src/pymc-devs/Theano-PyMC/theano/gof/cmodule.py", line 1229, in module_from_key
    module = lnk.compile_cmodule(location)
  File "/Users/dforeman/src/pymc-devs/Theano-PyMC/theano/gof/cc.py", line 1610, in compile_cmodule
    module = c_compiler.compile_str(
  File "/Users/dforeman/src/pymc-devs/Theano-PyMC/theano/gof/cmodule.py", line 2567, in compile_str
    raise Exception(
Exception: ('The following error happened while compiling the node', Elemwise{add,no_inplace}(InplaceDimShuffle{x,0}.0, <TensorType(float64, matrix)>), '\n', "Compilation failed (return status=1): /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:524:27: error: non-constant-expression cannot be narrowed from type 'npy_intp' (aka 'long') to 'int' in initializer list [-Wc++11-narrowing].     int init_totals[2] = {V5_n0, V3_n1};.                           ^~~~~. /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:524:27: note: insert an explicit cast to silence this issue.     int init_totals[2] = {V5_n0, V3_n1};.                           ^~~~~.                           static_cast<int>( ). /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:524:34: error: non-constant-expression cannot be narrowed from type 'npy_intp' (aka 'long') to 'int' in initializer list [-Wc++11-narrowing].     int init_totals[2] = {V5_n0, V3_n1};.                                  ^~~~~. /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:524:34: note: insert an explicit cast to silence this issue.     int init_totals[2] = {V5_n0, V3_n1};.                                  ^~~~~.                                  static_cast<int>( ). /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:536:12: error: non-constant-expression cannot be narrowed from type 'ssize_t' (aka 'long') to 'int' in initializer list [-Wc++11-narrowing].         0, V3_stride1, .            ^~~~~~~~~~. /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:536:12: note: insert an explicit cast to silence this issue.         0, V3_stride1, .            ^~~~~~~~~~.            static_cast<int>( ). /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:537:1: error: non-constant-expression cannot be narrowed from type 'ssize_t' (aka 'long') to 'int' in initializer list [-Wc++11-narrowing]. V5_stride0, V5_stride1, . ^~~~~~~~~~. /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:537:1: note: insert an explicit cast to silence this issue. V5_stride0, V5_stride1, . ^~~~~~~~~~. static_cast<int>( ). /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:537:13: error: non-constant-expression cannot be narrowed from type 'ssize_t' (aka 'long') to 'int' in initializer list [-Wc++11-narrowing]. V5_stride0, V5_stride1, .             ^~~~~~~~~~. /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:537:13: note: insert an explicit cast to silence this issue. V5_stride0, V5_stride1, .             ^~~~~~~~~~.             static_cast<int>( ). /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:538:1: error: non-constant-expression cannot be narrowed from type 'ssize_t' (aka 'long') to 'int' in initializer list [-Wc++11-narrowing]. V1_stride0, V1_stride1. ^~~~~~~~~~. /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:538:1: note: insert an explicit cast to silence this issue. V1_stride0, V1_stride1. ^~~~~~~~~~. static_cast<int>( ). /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:538:13: error: non-constant-expression cannot be narrowed from type 'ssize_t' (aka 'long') to 'int' in initializer list [-Wc++11-narrowing]. V1_stride0, V1_stride1.             ^~~~~~~~~~. /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:538:13: note: insert an explicit cast to silence this issue. V1_stride0, V1_stride1.             ^~~~~~~~~~.             static_cast<int>( ). 7 errors generated.. ", '[Elemwise{add,no_inplace}(<TensorType(float64, row)>, <TensorType(float64, matrix)>)]')
```

### 3. Theano Update

After seeing the latest pushes, I went ahead and only updated Theano PyMC
`pip install git+https://github.com/pymc-devs/Theano-PyMC`

Running the same code, I got this error from `import pymc3 as pm`

#### Error
```python
/anaconda3/envs/temp/lib/python3.8/site-packages/pymc3/theanof.py in <module>
     21 from theano.gof import Op
     22 from theano.gof.graph import inputs
---> 23 from theano.sandbox.rng_mrg import MRG_RandomStreams
     24 
     25 from .blocking import ArrayOrdering

ImportError: cannot import name 'MRG_RandomStreams' from 'theano.sandbox.rng_mrg' (/anaconda3/envs/temp/lib/python3.8/site-packages/theano/sandbox/rng_mrg.py)
```
## Versions and main components

* Theano version: theano-pymc  1.0.13+7.g9f7cf7d09
* Theano config (`python -c "import theano; print(theano.config)"`): Unknown
* Python version: 3.8: 3.8.5
* Operating system: [Clang 10.0.0 ] :: Anaconda, Inc. on darwin
* How did you install Theano: (conda/pip):
  * First it was from Pymc3, but then I did `pip install git+https://github.com/pymc-devs/Theano-PyMC`

**Comment by CloudChaoszero** [comment]
Sample of my `conda list`
```shell
numpy                     1.19.4           py38h6ced74f_2    conda-forge
olefile                   0.46               pyh9f0ad1d_1    conda-forge
openssl                   1.1.1i               h9ed2024_0  
opt_einsum                3.3.0                      py_0    conda-forge
packaging                 20.8               pyhd3deb0d_0    conda-forge
pandas                    1.0.5            py38h5fc983b_0    conda-forge
parso                     0.8.1              pyhd3eb1b0_0  
patsy                     0.5.1                      py_0    conda-forge
pexpect                   4.8.0              pyhd3eb1b0_3  
pickleshare               0.7.5           pyhd3eb1b0_1003  
pillow                    8.0.1            py38h565d989_0    conda-forge
pip                       20.3.3           py38hecd8cb5_0  
prompt-toolkit            3.0.8                      py_0  
protobuf                  3.12.3           py38hc84c608_0    conda-forge
ptyprocess                0.6.0              pyhd3eb1b0_2  
pygments                  2.7.3              pyhd3eb1b0_0  
pygpu                     0.7.6           py38h174b24a_1002    conda-forge
pymc3                     3.10.0             pyhd8ed1ab_0    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
python                    3.8.5                h26836e1_1  
python-dateutil           2.8.1                      py_0    conda-forge
python_abi                3.8                      1_cp38    conda-forge
pytz                      2020.4             pyhd8ed1ab_0    conda-forge
pyzmq                     20.0.0           py38h23ab428_1  
readline                  8.0                  h1de35cc_0  
scipy                     1.5.1            py38h38b60c6_0    conda-forge
setuptools                51.0.0           py38hecd8cb5_2  
six                       1.15.0             pyh9f0ad1d_0    conda-forge
sqlite                    3.33.0               hffcf06c_0  
tapi                      1000.10.8            h879752b_4    conda-forge
theano-pymc               1.0.13+7.g9f7cf7d09          pypi_0    pypi
tk                        8.6.10               hb0a8c7a_0  
tornado                   6.1              py38h7e3306e_0    conda-forge
traitlets                 5.0.5                      py_0  
typing-extensions         3.7.4.3                       0    conda-forge
typing_extensions         3.7.4.3                    py_0    conda-forge
wcwidth                   0.2.5                      py_0  
wheel                     0.36.2             pyhd3eb1b0_0  

```

**Comment by michaelosthege** [comment]
Looks like you're trying to combine PyMC3 3.10.0 with latest Theano-PyMC. This will not work, because we made some changes in Theano-PyMC that break some internals in PyMC3. That's why PyMC3 3.10.0 is pinned to Theano-PyMC 1.0.11.

**Comment by CloudChaoszero** [comment]
@michaelosthege Thanks for the response. Okay, so when I download using `conda install -c conda-forge pymc3`, step 1, I get an error exactly like #127. 

Example: Elemwise{add,no_inplace}
```python
Exception: ('The following error happened while compiling the node', Elemwise{add,no_inplace}(InplaceDimShuffle{x,0}.0, <TensorType(float64, matrix)>), '\n', "Compilation failed (return status=1): /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:524:27: error: non-constant-expression cannot be narrowed from type 'npy_intp' (aka 'long') to 'int' in initializer list [-Wc++11-narrowing].     int init_totals[2] = {V5_n0, V3_n1};.                           ^~~~~. /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:524:27: note: insert an explicit cast to silence this issue.     int init_totals[2] = {V5_n0, V3_n1};.                           ^~~~~.                           static_cast<int>( ). /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:524:34: error: non-constant-expression cannot be narrowed from type 'npy_intp' (aka 'long') to 'int' in initializer list [-Wc++11-narrowing].     int init_totals[2] = {V5_n0, V3_n1};.                                  ^~~~~. /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:524:34: note: insert an explicit cast to silence this issue.     int init_totals[2] = {V5_n0, V3_n1};.                                  ^~~~~.                                  static_cast<int>( ). /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:536:12: error: non-constant-expression cannot be narrowed from type 'ssize_t' (aka 'long') to 'int' in initializer list [-Wc++11-narrowing].         0, V3_stride1, .            ^~~~~~~~~~. /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:536:12: note: insert an explicit cast to silence this issue.         0, V3_stride1, .            ^~~~~~~~~~.            static_cast<int>( ). /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:537:1: error: non-constant-expression cannot be narrowed from type 'ssize_t' (aka 'long') to 'int' in initializer list [-Wc++11-narrowing]. V5_stride0, V5_stride1, . ^~~~~~~~~~. /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:537:1: note: insert an explicit cast to silence this issue. V5_stride0, V5_stride1, . ^~~~~~~~~~. static_cast<int>( ). /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:537:13: error: non-constant-expression cannot be narrowed from type 'ssize_t' (aka 'long') to 'int' in initializer list [-Wc++11-narrowing]. V5_stride0, V5_stride1, .             ^~~~~~~~~~. /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:537:13: note: insert an explicit cast to silence this issue. V5_stride0, V5_stride1, .             ^~~~~~~~~~.             static_cast<int>( ). /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:538:1: error: non-constant-expression cannot be narrowed from type 'ssize_t' (aka 'long') to 'int' in initializer list [-Wc++11-narrowing]. V1_stride0, V1_stride1. ^~~~~~~~~~. /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:538:1: note: insert an explicit cast to silence this issue. V1_stride0, V1_stride1. ^~~~~~~~~~. static_cast<int>( ). /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:538:13: error: non-constant-expression cannot be narrowed from type 'ssize_t' (aka 'long') to 'int' in initializer list [-Wc++11-narrowing]. V1_stride0, V1_stride1.             ^~~~~~~~~~. /Users/dforeman/.theano/compiledir_macOS-10.15.7-x86_64-i386-64bit-i386-3.8.5-64/tmpsatm5hgs/mod.cpp:538:13: note: insert an explicit cast to silence this issue. V1_stride0, V1_stride1.             ^~~~~~~~~~.             static_cast<int>( ). 7 errors generated.. ", '[Elemwise{add,no_inplace}(<TensorType(float64, row)>, <TensorType(float64, matrix)>)]')

```

Should I update theano, in that case?
______
I can re-try again too haha

**Comment by michaelosthege** [comment]
1st make sure that Theano is not already installed in the environment.
Then you could try installing pymc3 via pip. On Windows the install strategy through conda-forge gave me a lot of problems once, so I don't trust it anymore.

Conda runs some of the tests after install. Maybe the problem arises because of that?

**Comment by brandonwillard** [comment]
Looks like that new error is #127.
